src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2386,17 +2380,34 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     void TermControl::BellLightOn()
     {
+        // Initialize the animation if it does not exist
+        // We only initialize here instead of in the ctor because depending on the bell style setting,
+        // we may never need this animation
+        if (!_bellLightAnimation)
+        {
+            _bellLightAnimation = Window::Current().Compositor().CreateScalarKeyFrameAnimation();
+            // Add key frames and a duration to our bell light animation
+            _bellLightAnimation.InsertKeyFrame(0.0, 2.0);
+            _bellLightAnimation.InsertKeyFrame(1.0, 1.0);
+            _bellLightAnimation.Duration(winrt::Windows::Foundation::TimeSpan(std::chrono::milliseconds(TerminalWarningBellInterval)));
+        }
+
+        // Similar to the animation, only initialize the timer here
+        if (!_bellLightTimer)
+        {
+            DispatcherTimer invertTimer;
+            invertTimer.Interval(std::chrono::milliseconds(TerminalWarningBellInterval));
+            invertTimer.Tick({ get_weak(), &TermControl::_BellLightOff });
+            _bellLightTimer = invertTimer;

```

Comment:
Since you don't intend to further use `invertTimer` it's more efficient to do this:
```cpp
_bellLightTimer = std::move(invertTimer);
```

That way we don't increment the reference count in the assignment operator and then decrement it again when `invertTimer` is destroyed right after.

You can also write this though:
```cpp
_bellLightTimer = {};
_bellLightTimer.Interval(std::chrono::milliseconds(TerminalWarningBellInterval));
_bellLightTimer.Tick({ get_weak(), &TermControl::_BellLightOff });
```

Reply:
Yep makes sense, thank you!